### PR TITLE
 [doc] maintain document about sklearn

### PR DESCRIPTION
**Pull request by enori**
I suggest to maintain the document so that it reads more clearly to the reader.

## Why are these changes needed?

- It would be easier to understand if there was an explanation of the action of the methods Ray provides as well as how to use them.
- There are some broken links.
- Rather than a "Run a Cluster" section, why not add an explanation of the relationship between the sklearn process and the Ray cluster? I misunderstood at first.

## Related issue number

no.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

Since this PR does not affect the functionality of Ray itself, no specific testing has been done.

### CODE REVIEW

File: doc/source/ray-more-libs/joblib.rst [hunk 1]
```diff
@@ -6,9 +6,8 @@ Distributed Scikit-learn / Joblib
 .. _`issue on GitHub`: https://github.com/ray-project/ray/issues
 
 Ray supports running distributed `scikit-learn`_ programs by
-implementing a Ray backend for `joblib`_ using `Ray Actors <actors.html>`__
-instead of local processes. This makes it easy to scale existing applications
-that use scikit-learn from a single node to a cluster.
+implementing a Ray backend for `joblib`_ using `Ray Actors <../ray-core/actors.html>`__
+instead of local processes. This makes it easy to distribute existing application process that use scikit-learn from a single node to a cluster.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:10]: I prefer the existing wording here, I think talking about processes is too low level for this description. if we want to add it, we can add a more detailed "architecture" section

File: doc/source/ray-more-libs/joblib.rst [hunk 2]
```diff
@@ -21,13 +20,25 @@ that use scikit-learn from a single node to a cluster.
...
 ----------
 
-To get started, first `install Ray <installation.html>`__, then use
-``from ray.util.joblib import register_ray`` and run ``register_ray()``.
-This will register Ray as a joblib backend for scikit-learn to use.
-Then run your original scikit-learn code inside
-``with joblib.parallel_backend('ray')``. This will start a local Ray cluster.
-See the `Run on a Cluster`_ section below for instructions to run on
-a multi-node Ray cluster instead.
+`Install Ray <../ray-overview/installation.html>`__ and call some methods in your scikit-learn code as the following.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:23]:
```suggestion
`Install Ray <../ray-overview/installation.html>`__ and call some methods in your scikit-learn code as follows.
```

File: doc/source/ray-more-libs/joblib.rst [hunk 3]
```diff
@@ -21,13 +20,25 @@ that use scikit-learn from a single node to a cluster.
...
+  import joblib
+  from ray.util.joblib import register_ray
+  register_ray()
+  with joblib.parallel_backend('ray'):
+      # scikit-learn code
+
+Usage
+-----
+
+``from ray.util.joblib import register_ray`` imports joblib with Ray backend library.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:36]:
```suggestion
``from ray.util.joblib import register_ray`` imports the Ray backend for joblib.
```

File: doc/source/ray-more-libs/joblib.rst [hunk 4]
```diff
@@ -21,13 +20,25 @@ that use scikit-learn from a single node to a cluster.
...
+``register_ray()`` registers Ray as a joblib backend for scikit-learn to use.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:37]:
```suggestion
``register_ray()`` registers Ray as the joblib backend for scikit-learn to use.
```

Comment (edoakes) [reply doc/source/ray-more-libs/joblib.rst:37]: also, you can use backticks to format the library names like: `joblib` and `scikit-learn`

File: doc/source/ray-more-libs/joblib.rst [hunk 5]
```diff
@@ -21,13 +20,25 @@ that use scikit-learn from a single node to a cluster.
...
+
+The program inside ``with joblib.parallel_backend()`` is executed on the backend used by Parallel, and
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:39]:
```suggestion
The program inside ``with joblib.parallel_backend()`` is executed using the default backend registered for parallel execution and 
```

File: doc/source/ray-more-libs/joblib.rst [hunk 6]
```diff
@@ -21,13 +20,25 @@ that use scikit-learn from a single node to a cluster.
...
+
+Usage
+-----
+
+``from ray.util.joblib import register_ray`` imports joblib with Ray backend library.
+``register_ray()`` registers Ray as a joblib backend for scikit-learn to use.
+
+The program inside ``with joblib.parallel_backend()`` is executed on the backend used by Parallel, and 
+the program inside ``with joblib.parallel_backend('ray')`` is executed on Ray as the backend used by Parallel.
+Then, ``with joblib.parallel_backend('ray')`` deploys your code on a Ray cluster. Inside the call, run your original scikit-learn code.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:41]:
```suggestion
the program inside ``with joblib.parallel_backend('ray')`` is executed on Ray as the backend used by Parallel.
Then, ``with joblib.parallel_backend('ray')`` deploys your code on a Ray cluster. Inside the call, run your original scikit-learn code.
```

```suggestion
the program inside ``with joblib.parallel_backend('ray')`` is executed using the Ray backend for parallel execution. Your `scikit-learn` code will be distributed to actors across the cluster and executed in parallel.
```

File: doc/source/ray-more-libs/joblib.rst [hunk 7]
```diff
@@ -48,9 +59,20 @@ a multi-node Ray cluster instead.
   import joblib
   from ray.util.joblib import register_ray
   register_ray()
+  # ray.init(address=<address>) # if necessary
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:62]: you can drop this, it's implicitly called

File: doc/source/ray-more-libs/joblib.rst [hunk 8]
```diff
@@ -48,9 +59,20 @@ a multi-node Ray cluster instead.
...
   with joblib.parallel_backend('ray'):
       search.fit(digits.data, digits.target)
 
+This will start a local Ray cluster.
```
Comment (edoakes) [inline doc/source/ray-more-libs/joblib.rst:66]:
```suggestion
By default this will start a local Ray cluster. 
```

**Review by edoakes — COMMENTED** [review]
Thanks for the updates! Some grammatical and wording suggestions inline.

**Comment by enori** [comment]
thanks for your review.
I agree with your suggestion!

**Comment by enori** [comment]
Thanks for your activities!
If there are no problems, could you merge them?

**Pull request closed without merging**